pkg/runtime/scheme.go

```diff
@@ -153,6 +153,38 @@ func (s *Scheme) AddUnversionedTypes(version unversioned.GroupVersion, types ...
 	}
 }
 
+func validateListType(t reflect.Type) {
+	// exceptions
+	if t.Name() == "APIGroupList" || t.Name() == "APIResourceList" {
+		return
+	}
+	if name := t.Name(); len(name) >= 4 && name[len(name)-4:] == "List" {
+		var foundMetadata, foundItems bool
+		for i := 0; i < t.NumField(); i++ {
+			tag := t.Field(i).Tag.Get("json")
+			switch {
+			case len(tag) >= len("metadata") && tag[:len("metadata")] == "metadata":
```

Comment:
This is incorrect in general - why are we checking the presence of metadata.  Rules like "metadata" do not belong in `pkg/runtime`, they belong in `pkg/api/meta` at best.